.github/workflows/pr.yml

```diff
@@ -2,27 +2,86 @@ name: Pull Request
 
 on: 
   workflow_dispatch: {}
+  pull_request: {}
+  push:
+    branches:
+      - main
 
 jobs:
   build:
     strategy:
-      matrix:
-        javaVersion: [8, 11, 16, 17]
-        os: [ubuntu-latest, windows-latest]
       fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        versions: [
+            { java: 8, gradle: 6.8 },
+            { java: 11, gradle: 6.8 },
+            { java: 16, gradle: 7.3.1 },
+            { java: 17, gradle: 7.3.1 }
+        ]
     runs-on: ${{ matrix.os }}
     steps:
     - uses: actions/checkout@v4
 
     - uses: actions/setup-java@v3
       with:
         distribution: temurin
-        java-version: ${{ matrix.javaVersion}}
+        java-version: ${{ matrix.versions.java}}
         
     - name: Setup Gradle
       uses: gradle/gradle-build-action@v2
       with:
-        gradle-version: 6.8
+        gradle-version: ${{ matrix.versions.gradle }}
+
+    - name: Setup gradle.properties
```

Comment:
I should check again whether this is still needed. I changed up a bunch of stuff since I added this, but at one point it was needed in order to get java 16+17 building.